migrations/mysql/20221109092556_torrust_user_date_registered_allow_null.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE torrust_users CHANGE date_registered date_registered DATETIME NOT NULL
```

Comment:
This table alteration sets `NOT NULL`, shouldn't it be `DEFAULT NULL`?
Strange that a test didn't break with this...

Reply:
Hi @da2ce7, good catch! I've changed only two things in the application behaviour:

1. The login works with both "argon2" and "pbkdf2". I tested it manually.
2. The "date_registered" is now optional (It can be NULL). I only tested the MySQL migrations by running manually "cargo test", but I did not test to create a new user without that date in MySQL because there is no migration for MySQL. There was no MySQL database in version v1.0.0. We could keep that NOT NULL option for MySQL, but I have to fix it to have the same schema as in SQLite.

Reply:
I've added two new subtasks to the main PR description to test those changes.